usb/esp_tinyusb/tusb_msc_storage.c

```diff
@@ -0,0 +1,359 @@
+static const char *s_base_path;
+
+static const char *TAG = "tinyusb_msc_storage";
+
+static esp_err_t _spiflash_init(const tinyusb_msc_storage_config_t *config)
+{
+    ESP_LOGI(TAG, "Initializing wear levelling");
+
+    const esp_partition_t *data_partition = esp_partition_find_first(ESP_PARTITION_TYPE_DATA, ESP_PARTITION_SUBTYPE_DATA_FAT, NULL);
+    if (data_partition == NULL) {
+        ESP_LOGE(TAG, "Failed to find FATFS partition. Check the partition table.");
+        return ESP_ERR_NOT_FOUND;
+    }
+
+    return wl_mount(data_partition, &s_wl_handle);
+}
+
+#if SOC_SDMMC_HOST_SUPPORTED
+static esp_err_t _sdmmc_init(const tinyusb_msc_storage_config_t *config)
+{
+    esp_err_t ret = ESP_OK;
+    bool host_init = false;
+
+    ESP_LOGI(TAG, "Initializing SDCard");
+
+    sdmmc_host_t host = config->sdmmc_config->host;
+    sdmmc_slot_config_t slot_config = config->sdmmc_config->slot_config;
+
+    // not using ff_memalloc here, as allocation in internal RAM is preferred
+    s_card = (sdmmc_card_t *)malloc(sizeof(sdmmc_card_t));
+    ESP_GOTO_ON_FALSE(s_card, ESP_ERR_NO_MEM, clean, TAG, "could not allocate new sdmmc_card_t");
+
+    ESP_GOTO_ON_ERROR((*host.init)(), clean, TAG, "Host Config Init fail");
+    host_init = true;
+
+    ESP_GOTO_ON_ERROR(sdmmc_host_init_slot(host.slot, (const sdmmc_slot_config_t *) &slot_config),
+                      clean, TAG, "Host init slot fail");
+
+    while (sdmmc_card_init(&host, s_card)) {
+        ESP_LOGE(TAG, "The detection pin of the slot is disconnected(Insert uSD card). Retrying...");
+        vTaskDelay(pdMS_TO_TICKS(3000));
+    }
+
+    // Card has been initialized, print its properties
+    sdmmc_card_print_info(stdout, s_card);
+
+    return ESP_OK;
+
+clean:
+    if (host_init) {
+        if (host.flags & SDMMC_HOST_FLAG_DEINIT_ARG) {
+            host.deinit_p(host.slot);
+        } else {
+            (*host.deinit)();
+        }
+    }
+    if (s_card) {
+        free(s_card);
+        s_card = NULL;
+    }
+    return ret;
+}
+#endif
+
+esp_err_t tinyusb_msc_storage_init(const tinyusb_msc_storage_type_t storage_type, const tinyusb_msc_storage_config_t *config)
+{
+    esp_err_t ret = ESP_FAIL;
+    switch (storage_type) {
+    case TINYUSB_MSC_STORAGE_SPI:
+        ret = _spiflash_init(config);
+        break;
+#if SOC_SDMMC_HOST_SUPPORTED
+    case TINYUSB_MSC_STORAGE_SDMMC:
+        ret = _sdmmc_init(config);
+        break;
+#endif
+    default:
+        ESP_LOGE(TAG, "Invalid storage type = %d", storage_type);
+        ret = ESP_FAIL;
+        break;
+    }
+    return ret;
+}
+
+static inline size_t esp_vfs_fat_get_allocation_unit_size(
+    size_t sector_size, size_t requested_size)
+{
+    size_t alloc_unit_size = requested_size;
+    const size_t max_sectors_per_cylinder = 128;
+    const size_t max_size = sector_size * max_sectors_per_cylinder;
+    alloc_unit_size = MAX(alloc_unit_size, sector_size);
+    alloc_unit_size = MIN(alloc_unit_size, max_size);
```

Comment:
IMHO this is not so clear to the developer what is happening here. 
Probably, the changes and a ternary operator could help us to make it a little bit better:

    alloc_unit_size = MAX(requested_size, sector_size);
    alloc_unit_size = (alloc_unit_size < max_size)? alloc_unit_size : max_size;

With such code we also can more easily (and clearer to the developer) cover the boundary conditions when alloc_unit_size is equal to max_size.

Reply:
The function definition was redundant, removed.